mod(security): Security, OAuth2 의 Controller, Service, Mapper 수정 및 작성.

LancerAlert opened the pull request.

3ll3702 committed:
mod(security): rebase develop

3ll3702 committed:
Update UserDTO.java

LancerAlert committed:
feature(security): rebase develop

LancerAlert committed:
feature(security): 기본 틀 작성.

LancerAlert committed:
feature(security): 기본 틀 작성.

LancerAlert committed:
mod(security): Security, OAuth2 의 Controller, Service, Mapper 수정 및 작성.

LancerAlert committed:
mod(security): Security, OAuth2 의 Controller, Service, Mapper 수정 및 작성.

LancerAlert committed:
mod(security): Security, OAuth2 의 Controller, Service, Mapper 수정 및 작성.

LancerAlert committed:
mod(security): OAuth2 의 클라이언트 키가 설정되지 않아 실행되지 않던 오류 수정.

LancerAlert committed:
fix(security): OAuth2 의 클라이언트 설ì  키가 없어 발생한 오류 해결.

LancerAlert committed:
mod(security): 중복 클래스 제거.

LancerAlert committed:
Merge branch 'back/security' of https://github.com/mj2sdev/yorimichi into back/security

LancerAlert committed:
mod(security): 현재 브랜치에 불필요한 파일 제거.

LancerAlert committed:
mod(security): Security 구성 수정.
- SecurityConfig: Manager 직접 주입 수정.
- AuthController: SecurityConfig 에 맞게 수정.
- Security 및 Login, SocialLogin 흐름에 대한 문서 추가.

mj2sdev commented on src/main/java/com/jslhrd/yorimichi/config/SecurityConfig.java:
```diff
@@ -1,23 +1,235 @@
+	public WebSecurityCustomizer webSecurityCustomizer() {
+		return web -> web.ignoring().requestMatchers(
+				"/favicon.ico",
+				"/assets/**", "/css/**", "/js/**", "/images/**",
```
저희 프로젝트 이미지 폴더 images 맞음?

mj2sdev commented on src/main/java/com/jslhrd/yorimichi/security/AppUserPrincipal.java:
```diff
@@ -0,0 +1,207 @@
+
+	private static String normEmail(String email) {
+		return (email == null) ? null : email.trim().toLowerCase();
+	}
```
nomalizationEmail 어떻습니까

mj2sdev commented on src/main/java/com/jslhrd/yorimichi/service/manager/UserManager.java:
```diff
@@ -1,46 +1,45 @@
+	}
+
+	@Override
+	public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
```
분리되었으니 요건 이제 없어도 될듯합니다.

mj2sdev reviewed (approved):
많은 작업 고생많으셨습니다.
이제 실제 작동하는지 확인해보겠습니다.

LancerAlert committed:
mod(security): SecurityConfig 설정에 맞는 경로로 수정.

LancerAlert committed:
mod(security): Account, User 의 역할 구분. Service, Manager, Mapper 등에 맞춰 수정.

LancerAlert committed:
mod(security): User 수정 시 발생한 Account 수정사항.
- AccountManager, AccountService, AuthController: isNicknameAvailable 조회 메서드를 UserManager 로 이동으로 인한 수정.

LancerAlert committed:
mod(security): Account 와 User 의 역할을 구분하여 수정.
- User 는 조회 및 유저 데이터를 담당하게 수정.
- 조회 상세 데이터는 추후 추가.

LancerAlert committed:
mod(security): root insert 시 id 검증 추가.

LancerAlert committed:
mod(security): root insert 시 id 검증 추가.

LancerAlert committed:
Merge branch 'develop' into back/security

mj2sdev commented on src/main/java/com/jslhrd/yorimichi/service/manager/UserManager.java:
```diff
@@ -2,44 +2,71 @@
+				log.debug("User: update no-op userId={}", userId);
+				return;
+			}
+		} catch (DuplicateNicknameException e) {
```
아오 ㅋㅋ

mj2sdev reviewed (approved):
작업 고생하셨습니다.
